### ISSUE TITLE
🐛 bf16 matmul produces catastrophic error with large reduction dimension K (even with fp32_dest_acc_en)

### ISSUE THREAD

**Issue by nvukobratTT** [nvukobratTT/tt-xla#1]
## Problem
bf16 matmul precision degrades linearly with reduction dimension K on TT Blackhole, even with `fp32_dest_acc_en=true` and `math_fidelity=hifi4`. At K=8960 (used by Wan2.1 FFN), 82.8% of output elements are off by >1.0 vs CPU reference.

This makes **any model with FFN hidden_dim > ~4K produce incorrect output** (Wan2.1-1.3B ffn=8960, LLaMA ffn=11008, etc).

## Root Cause Analysis
Error scales ~linearly with K:

| K | max_diff | mean_diff | % elements > 1.0 |
|---|---|---|---|
| 128 | 0.25 | 0.006 | 0.0% |
| 512 | 0.50 | 0.055 | 0.0% |
| 1536 | 2.00 | 0.294 | 0.2% |
| 4096 | 8.00 | 1.279 | 48.4% |
| **8960** | **28.00** | **4.073** | **82.8%** |

**Critically:** `nn.Linear` with the same K=8960 weights gives max_diff=0.203 (fine!). The raw `@` / `torch.matmul` operator takes a different, broken code path.

## Impact
- Single TT transformer block produces cosine_sim=0.32 vs CPU (catastrophic)
- Even 1 block out of 30 on TT destroys pipeline output
- No hybrid CPU/TT workaround is viable
- `fp32_dest_acc_en=true` flag appears to not reach the raw matmul kernels (but does reach nn.Linear)

## Minimal Reproduction
```python
"""repro_matmul_large_k.py — run inside tt-xla-dev container"""
import torch, torch_xla, torch_xla.core.xla_model as xm, torch_xla.runtime as xr

xr.set_device_type("TT")
torch_xla.set_custom_compile_options({
    "fp32_dest_acc_en": "true", "math_fidelity": "hifi4",
})
device = xm.xla_device()
torch.manual_seed(42)

M, N = 768, 1536
for K in [128, 512, 1536, 4096, 8960]:
    A = torch.randn(1, M, K, dtype=torch.bfloat16)
    B = torch.randn(K, N, dtype=torch.bfloat16)
    cpu_out = A @ B
    tt_out = (A.to(device) @ B.to(device)); xm.mark_step()
    diff = (tt_out.cpu().float() - cpu_out.float()).abs()
    pct = (diff > 1.0).float().mean().item() * 100
    print(f"K={K:>5}: max={diff.max().item():.1f}, mean={diff.mean().item():.3f}, >1.0={pct:.1f}%")

# nn.Linear is fine at same K!
linear = torch.nn.Linear(8960, 1536, bias=False, dtype=torch.bfloat16)
x = torch.randn(1, 768, 8960, dtype=torch.bfloat16)
with torch.no_grad():
    cpu_ref = linear(x)
    tt_ref = linear.to(device)(x.to(device)); xm.mark_step()
diff = (tt_ref.cpu().float() - cpu_ref.float()).abs()
print(f"\nnn.Linear K=8960: max={diff.max().item():.3f}, >1.0={((diff>1).float().mean()*100):.1f}%")
```

**Expected:** max_diff < 1.0 for all K with fp32_dest_acc_en=true
**Actual:** max_diff=28.0 at K=8960, 82.8% elements >1.0

## Hypothesis
`fp32_dest_acc_en` may not be reaching raw matmul tile kernels — only `nn.Linear`/`F.linear` paths. Or tile-level partial accumulations happen in bf16 before a final f32 reduction.

## Files
- Full repro: `repro_matmul_large_k.py`
- K-dimension sweep: `wan_matmul_sweep.py`
- Workaround report: `docs/wan_workaround_results.md`
- Block dissection: `wan_block_dissect2.py`

## Workaround
None viable on TT. Full CPU inference required until fixed.

## Priority
**P0 Critical** — blocks all model inference with FFN dim > 4K.
